utils/s2n_safety.h

```diff
@@ -102,8 +102,8 @@ static inline void* trace_memcpy_check(void *restrict to, const void *restrict f
 #define GUARD_NONBLOCKING( x )          \
   do {                                  \
     int __tmp_r = (x);                  \
-    if (ERR_IS_BLOCKING( __tmp_r )) {   \
-      S2N_ERROR( __tmp_r );             \
+    if (__tmp_r == S2N_CALLBACK_BLOCKED) {   \
+      S2N_ERROR( S2N_ERR_APPLICATION_BLOCKED );             \
```

Comment:
I can see wanting to return an internal error code here. This does modify the behavior of async functions though, they will always return S2N_ERR_APPLICATION_BLOCKED instead of the original error code.

Reply:
Yes you are right. What I want to do here is to translate the return code to an internal error code.